File: Beecrowd_Parte1/Cordão_de_Led_3171.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 101

// Lista de adjacência
int adjList[MAX][MAX];
// Número de vizinhos de cada nó
int neighborsCount[MAX];
// Array de visitados
int visited[MAX];

// Função de busca em profundidade (DFS)
void dfs(int node) {
    visited[node] = 1;
    for (int i = 0; i < neighborsCount[node]; i++) {
        int neighbor = adjList[node][i];
        if (!visited[neighbor]) {
            dfs(neighbor);
        }
    }
}

int main() {
    int N, L;
    scanf("%d %d", &N, &L);
    
    // Inicializando a lista de adjacências e o array de vizinhos
    for (int i = 1; i <= N; i++) {
        neighborsCount[i] = 0;
    }
    
    // Lendo as ligações e populando a lista de adjacências
    for (int i = 0; i < L; i++) {
        int X, Y;
        scanf("%d %d", &X, &Y);
        adjList[X][neighborsCount[X]++] = Y;
        adjList[Y][neighborsCount[Y]++] = X;
    }
    
    // Inicializando o array de visitados
    for (int i = 1; i <= N; i++) {
        visited[i] = 0;
    }
    
    // Realizando a DFS a partir do nó 1
    dfs(1);
    
    // Verificando se todos os nós foram visitados
    int completo = 1;
    for (int i = 1; i <= N; i++) {
        if (!visited[i]) {
            completo = 0;
            break;
        }
    }
    
    // Imprimindo o resultado
    if (completo) {
        printf("COMPLETO\n");
    } else {
        printf("INCOMPLETO\n");
    }
    
    return 0;
}

File: Beecrowd_Parte1/Cordão_de_Led_3171.java
import java.util.ArrayList;
import java.util.Scanner;

public class Cordão_de_Led_3171 {
    static ArrayList<Integer>[] adjList;
    static boolean[] visited;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Lendo N (número de segmentos) e L (número de ligações)
        int N = scanner.nextInt();
        int L = scanner.nextInt();
        
        // Inicializando a lista de adjacências
        adjList = new ArrayList[N + 1];
        for (int i = 1; i <= N; i++) {
            adjList[i] = new ArrayList<>();
        }
        
        // Lendo as ligações e populando a lista de adjacências
        for (int i = 0; i < L; i++) {
            int X = scanner.nextInt();
            int Y = scanner.nextInt();
            adjList[X].add(Y);
            adjList[Y].add(X);
        }
        
        // Inicializando o array de visitados
        visited = new boolean[N + 1];
        
        // Realizando a DFS a partir do nó 1
        dfs(1);
        
        // Verificando se todos os nós foram visitados
        boolean completo = true;
        for (int i = 1; i <= N; i++) {
            if (!visited[i]) {
                completo = false;
                break;
            }
        }
        
        // Imprimindo o resultado
        if (completo) {
            System.out.println("COMPLETO");
        } else {
            System.out.println("INCOMPLETO");
        }
        
        scanner.close();
    }
    
    // Método para realizar a busca em profundidade (DFS)
    static void dfs(int node) {
        visited[node] = true;
        for (int neighbor : adjList[node]) {
            if (!visited[neighbor]) {
                dfs(neighbor);
            }
        }
    }
}
